rasterio/_base.pyx

```diff
@@ -13,6 +13,7 @@ from rasterio._err import (
     GDALError, CPLE_BaseError, CPLE_IllegalArgError, CPLE_OpenFailedError,
     CPLE_NotSupportedError)
 from rasterio._err cimport exc_wrap_pointer, exc_wrap_int
+from rasterio._shim cimport open_dataset
```

Comment:
`gdal_dataset` is our new function that covers up the differences between GDAL 1.x and 2.x.